src/katgpucbf/xbgpu/engine.py

```diff
@@ -282,23 +279,24 @@ def __init__(
         super().__init__(katcp_host, katcp_port)
         self._cancel_tasks: list[asyncio.Task] = []  # Tasks that need to be cancelled on shutdown
 
+        # B-engine doesn't work yet
+        assert len(outputs) == 1
+        assert isinstance(outputs[0], XOutput)
```

Comment:
My feeling is that assertions are fine if you believe that it can never fail, no matter what the user does, and you want to document that belief (and the computer telling you that your believe is wrong is a bonus). That way if it's eliminated by `python -O` you don't really care.

I don't think that's applicable in this case, because if the user doesn't pass `--corrprod`, this assertion will fail. So it would be a little better to replace this with an exception that explains that `--corrprod` is required. But also as James says, this is a stepping stone and we don't have to make things perfect if we know we're about to rip them out anyway.